hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -156,6 +156,9 @@ protected void changeSupply(
         copyToken.totalSupply(newTotalSupply);
         copyTreasuryRel.balance(newTreasuryBalance);
 
+        // We also need to update the Titles count on the copy
+        copyTreasuryAccount.numberTreasuryTitles(treasuryAccount.numberTreasuryTitles() + 1);
```

Comment:
This method is only meant to change the supply. Why would we increment the treasury titles on a token burn?

Reply:
In CryptoDeleteSuite,  validateFalse(deletedAccount.numberTreasuryTitles() > 0, ACCOUNT_IS_TREASURY); is checking if this is set in test cannotDeleteTreasuryAccount. During token create, if we set the treasury, we need to update this value. It's hard to explain in a couple sentences. This is the the number of tokens for which this account is treasury. This change made the test pass as well.

Reply:
Yes, `numberTreasuryTitles` needs to increase in a token create, so this code should be inside of the token create handler. It shouldn't be in this method because this method is also used for burns

Reply:
Removed the code change to TokenCreateHandler.